rts/Map/Ground.cpp

```diff
@@ -461,7 +461,7 @@ const float* CGround::GetApproximateHeightUnsafePtr(int x, int z, bool synced)
 float CGround::GetHeightAboveWater(float x, float z, bool synced)
 {
 	RECOIL_DETAILED_TRACY_ZONE;
-	return std::max(0.0f, GetHeightReal(x, z, synced));
+	return std::max(GetWaterLevel(x, z, synced), GetHeightReal(x, z, synced));
```

Comment:
I'm not entirely sure about this one. The comment on this function in the header file basically says this should return a non-negative value and doesn't mention water at all.

If this was used in places that weren't doing water related things, but instead just using it to ensure a non-negative value then this could cause problems down the line once dynamic water is implemented. Right now with this just being the equivalent of a hard coded 0 with extra steps it should behave the same, but part of doing this work seems to be to make it so water doesn't have to be a hard coded 0.

Reply:
If in doubt, leave it out.
```suggestion
	return std::max(0.0f, GetHeightReal(x, z, synced)); // FIXME GetWaterLevel, but has non-water use cases!
```

Reply:
That gets used in a lot of places, though the uses I've seen is for: height, no lower than the water level. In that sense the fucntion name is misleading. The issue here, with the function being used that way, code working on that assumption will not behave correctly with the water level is changed.

Reply:
Yeah, I'm certain this change here won't cause any problems now. And based on the name of the method, this is going to return the correct value.

It's just that the comment on this smelled of "Something is probably relying on the current effect rather than the actual semantic of wanting the height above water specifically". That should come out later when dynamic water is actually being implemented, and I don't want to leave a FIXME comment behind that would be easy to miss later (some of the ones I saw in those codebase are VERY old already).

Reply:
Actually, I had forgotten to push a change I'd made to this after looking at it for a while 😅 

The new version should reflect accurate math for calculating the distance above the water level while still return the exact same values this function used to until dynamic water is implemented.